## 57% of end-end tests have passed
```
Total executed tests: 7
Failed tests: 3
Expected to fail tests: 0
Passed tests: 4
```
```
IDs of failed tests: 727230,702745,703133 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=20446-collectibles-activity-tab&PR_ID=21109&APK_NAME=21109.apk&TR_CASE_IDS=727230,702745,703133">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659732">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><p><blockquote><div>Device 2: Accepting incoming CR for sender </div><div>Device 2: Find `Button` by `xpath`: `//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` </div></blockquote></p>

```
Test setup failed: ../views/home_view.py:409: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:162: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:159: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:164: in wait_for_rendering_ended_and_click
    self.click()
../views/base_element.py:92: in click
    element.click()
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:94: in click
    self._execute(Command.CLICK_ELEMENT)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute
    return self._parent.execute(command, params)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: Could not proxy command to the remote server. Original error: socket hang up
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.commands.click (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/element.js:81:10)

During handling of the above exception, another exception occurred:
critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:421: in handle_contact_request
    self.close_activity_centre.wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at processTicksAndRejections (node:internal/process/task_queues:95:5)
E       at AndroidUiautomator2Driver.helpers.doFindElementOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/lib/commands/find.js:30:10)
E       at doFind (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:36:17)
E       at wrappedCondFn (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/mnt/sauce/appium/appium-v2.0.0/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/mnt/sauce/appium/appium-v2.0.0/appium_home/linux/node_modules/appium-uiautomator2-driver/node_modules/appium-android-driver/lib/commands/find.js:60:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/mnt/sauce/appium/appium-v2.0.0/packages/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/459ede6f0fc941ab86816d7ed395ddaf?auth=9cb8e2f77ab9bb0a5a47aec93ff6c56c">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f8510bfea5004a1591d8f1313fb12d61?auth=b69dfcba8a67b05390d8281f2e19835d">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659736">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Text` by `xpath`: `//android.view.ViewGroup[@content-desc='container']/android.widget.TextView[@text='Ether']/../android.widget.TextView[3]` </div><div>Device 2: `Text` is `0.04859 ETH` </div></blockquote></p>

```
critical/test_wallet.py:190: in test_wallet_send_asset_from_drawer
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Sender balance is not updated on Etherscan, it is 0.4387 but expected to be 0.4388
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659734">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Type `Restore user` to `EditBox` </div><div>Device 1: Click until `EditBox` by `xpath`: `//*[@text="Type password"]` will be presented </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:160: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.sign_in.recover_access(passphrase=waku_user.seed, second_user=True)
../views/sign_in_view.py:295: in recover_access
    self.set_profile(username, set_image)
../views/sign_in_view.py:232: in set_profile
    self.profile_continue_button.click_until_presence_of_element(self.profile_password_edit_box)
../views/base_element.py:107: in click_until_presence_of_element
    while not desired_element.is_element_displayed(1) and counter <= attempts:
../views/base_element.py:221: in is_element_displayed
    return self.wait_for_visibility_of_element(sec, ignored_exceptions=ignored_exceptions)
../views/base_element.py:145: in wait_for_visibility_of_element
    .until(expected_conditions.visibility_of_element_located((self.by, self.locator)))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:86: in until
    value = method(self._driver)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/support/expected_conditions.py:152: in _predicate
    return _element_if_visible(driver.find_element(*locator))
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs-rerun@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 The test session has already finished, and can't receive further commands. For help please check https://docs.saucelabs.com/dev/error-messages
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5ce2fc3d7c9e42b8a5c2a054ee351cf3?auth=2cd8841f2db76561dc0b15466828dad2">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (4)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659735">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659733">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/04e0a31b7b744805a5490a2b843a031a?auth=f875cd66bbf06a47cc6de38fc8130e77#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/027bcf8a73154dfcadaf1eb2124c28c2?auth=11dd2be475dde8e28865319dd11cfee5#107">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659731">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5ce2fc3d7c9e42b8a5c2a054ee351cf3?auth=2cd8841f2db76561dc0b15466828dad2">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3659737">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/05b75c68b15d4d3fbe96a9bf199fc0dc?auth=d2cc74c9bc520243ec45cccfe820d27c#39">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/quo/components/tags/context_tag/style.cljs
+++ b/src/quo/components/tags/context_tag/style.cljs
@@ -60,8 +60,12 @@
     shrinkable? (assoc :flex-shrink 1)))
 
 (defn text
-  [theme]
-  {:color (colors/theme-colors colors/neutral-100 colors/white theme)})
+  ([theme]
+   (text theme false))
+  ([theme gray-text?]
+   {:color (if gray-text?
+             colors/neutral-50
+             (colors/theme-colors colors/neutral-100 colors/white theme))}))
 
 (defn token-logo
   [size]

--- a/src/quo/components/tags/context_tag/view.cljs
+++ b/src/quo/components/tags/context_tag/view.cljs
@@ -1,21 +1,21 @@
 (ns quo.components.tags.context-tag.view
   (:require
-   [clojure.string :as string]
-   [quo.components.avatars.account-avatar.view :as account-avatar]
-   [quo.components.avatars.group-avatar.view :as group-avatar]
-   [quo.components.avatars.user-avatar.view :as user-avatar]
-   [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
-   [quo.components.icon :as icons]
-   [quo.components.list-items.preview-list.view :as preview-list]
-   [quo.components.markdown.text :as text]
-   [quo.components.tags.context-tag.schema :as component-schema]
-   [quo.components.tags.context-tag.style :as style]
-   [quo.components.utilities.token.view :as token]
-   [quo.foundations.colors :as colors]
-   [quo.theme :as quo.theme]
-   [react-native.core :as rn]
-   [react-native.fast-image :as fast-image]
-   [schema.core :as schema]))
+    [clojure.string :as string]
+    [quo.components.avatars.account-avatar.view :as account-avatar]
+    [quo.components.avatars.group-avatar.view :as group-avatar]
+    [quo.components.avatars.user-avatar.view :as user-avatar]
+    [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
+    [quo.components.icon :as icons]
+    [quo.components.list-items.preview-list.view :as preview-list]
+    [quo.components.markdown.text :as text]
+    [quo.components.tags.context-tag.schema :as component-schema]
+    [quo.components.tags.context-tag.style :as style]
+    [quo.components.utilities.token.view :as token]
+    [quo.foundations.colors :as colors]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]
+    [react-native.fast-image :as fast-image]
+    [schema.core :as schema]))
 
 (defn- tag-skeleton
   [{:keys [size text theme shrinkable? gray-text?]
@@ -168,9 +168,10 @@
             :gray-text?  gray-text?}
            (if nft-placeholder?
              [icons/icon :i/nft {:size 20}]
-             [rn/image {:style    (style/rounded-logo size)
-                        :source   collectible
-                        :on-error #(set-image-error true)}])])
+             [rn/image
+              {:style    (style/rounded-logo size)
+               :source   collectible
+               :on-error #(set-image-error true)}])])
 
         :account
         [tag-skeleton {:theme theme :size size :text account-name}

--- a/src/quo/components/tags/context_tag/view.cljs
+++ b/src/quo/components/tags/context_tag/view.cljs
@@ -1,31 +1,33 @@
 (ns quo.components.tags.context-tag.view
   (:require
-    [quo.components.avatars.account-avatar.view :as account-avatar]
-    [quo.components.avatars.group-avatar.view :as group-avatar]
-    [quo.components.avatars.user-avatar.view :as user-avatar]
-    [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
-    [quo.components.icon :as icons]
-    [quo.components.list-items.preview-list.view :as preview-list]
-    [quo.components.markdown.text :as text]
-    [quo.components.tags.context-tag.schema :as component-schema]
-    [quo.components.tags.context-tag.style :as style]
-    [quo.components.utilities.token.view :as token]
-    [quo.foundations.colors :as colors]
-    [quo.theme :as quo.theme]
-    [react-native.core :as rn]
-    [react-native.fast-image :as fast-image]
-    [schema.core :as schema]))
+   [clojure.string :as string]
+   [quo.components.avatars.account-avatar.view :as account-avatar]
+   [quo.components.avatars.group-avatar.view :as group-avatar]
+   [quo.components.avatars.user-avatar.view :as user-avatar]
+   [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
+   [quo.components.icon :as icons]
+   [quo.components.list-items.preview-list.view :as preview-list]
+   [quo.components.markdown.text :as text]
+   [quo.components.tags.context-tag.schema :as component-schema]
+   [quo.components.tags.context-tag.style :as style]
+   [quo.components.utilities.token.view :as token]
+   [quo.foundations.colors :as colors]
+   [quo.theme :as quo.theme]
+   [react-native.core :as rn]
+   [react-native.fast-image :as fast-image]
+   [schema.core :as schema]))
 
 (defn- tag-skeleton
-  [{:keys [size text theme shrinkable?]
-    :or   {size  24
-           theme (quo.theme/use-theme)}}
+  [{:keys [size text theme shrinkable? gray-text?]
+    :or   {size       24
+           gray-text? false
+           theme      (quo.theme/use-theme)}}
    logo-component]
   [rn/view {:style (style/tag-container size)}
    logo-component
    [rn/view {:style (style/tag-spacing size shrinkable?)}
     [text/text
-     {:style           (style/text theme)
+     {:style           (style/text theme gray-text?)
       :weight          :medium
       :size            (if (= size 24) :paragraph-2 :paragraph-1)
       :number-of-lines 1
@@ -94,92 +96,106 @@
            state               :default
            theme               (quo.theme/use-theme)}
     :as   props}]
-  [rn/view {:style (merge {:align-items :flex-start} container-style)}
-   [rn/view
-    {:style               (style/container {:theme               theme
-                                            :type                type
-                                            :size                size
-                                            :state               state
-                                            :blur?               blur?
-                                            :customization-color customization-color})
-     :accessibility-label :context-tag}
-    (case type
-      :default
-      [tag-skeleton {:theme theme :size size :text full-name}
-       [user-avatar/user-avatar
-        {:full-name           full-name
-         :profile-picture     profile-picture
-         :size                (if (= size 24) :xxs 28)
-         :status-indicator?   false
-         :ring?               false
-         :customization-color customization-color}]]
+  (let [[image-error? set-image-error] (rn/use-state false)]
+    [rn/view {:style (merge {:align-items :flex-start} container-style)}
+     [rn/view
+      {:style               (style/container {:theme               theme
+                                              :type                type
+                                              :size                size
+                                              :state               state
+                                              :blur?               blur?
+                                              :customization-color customization-color})
+       :accessibility-label :context-tag}
+      (case type
+        :default
+        [tag-skeleton {:theme theme :size size :text full-name}
+         [user-avatar/user-avatar
+          {:full-name           full-name
+           :profile-picture     profile-picture
+           :size                (if (= size 24) :xxs 28)
+           :status-indicator?   false
+           :ring?               false
+           :customization-color customization-color}]]
 
-      :multiuser
-      [preview-list/view {:type :user :size :size-20}
-       users]
+        :multiuser
+        [preview-list/view {:type :user :size :size-20}
+         users]
 
-      :multinetwork
-      [preview-list/view {:type :network :size :size-20}
-       networks]
+        :multinetwork
+        [preview-list/view {:type :network :size :size-20}
+         networks]
 
-      :audio
-      [tag-skeleton {:theme theme :text (str duration)}
-       [rn/view {:style (style/audio-tag-icon-container customization-color theme)}
-        [icons/icon :i/play {:color style/audio-tag-icon-color :size 12}]]]
+        :audio
+        [tag-skeleton {:theme theme :text (str duration)}
+         [rn/view {:style (style/audio-tag-icon-container customization-color theme)}
+          [icons/icon :i/play {:color style/audio-tag-icon-color :size 12}]]]
 
-      :group
-      [tag-skeleton {:theme theme :size size :text group-name}
-       [group-avatar/view
-        {:icon-name           :i/members
-         :size                (if (= size 24) :size-20 :size-28)
-         :customization-color (colors/custom-color customization-color 50)}]]
+        :group
+        [tag-skeleton {:theme theme :size size :text group-name}
+         [group-avatar/view
+          {:icon-name           :i/members
+           :size                (if (= size 24) :size-20 :size-28)
+           :customization-color (colors/custom-color customization-color 50)}]]
 
-      (:channel :community)
-      [communities-tag (assoc props :channel? (= type :channel))]
+        (:channel :community)
+        [communities-tag (assoc props :channel? (= type :channel))]
 
-      :token
-      [tag-skeleton {:theme theme :size size :text (str amount " " token)}
-       [token/view
-        {:style (style/token-logo size)
-         :token token
-         :size  (if (= size 24) :size-20 :size-28)}]]
+        :token
+        [tag-skeleton {:theme theme :size size :text (str amount " " token)}
+         [token/view
+          {:style (style/token-logo size)
+           :token token
+           :size  (if (= size 24) :size-20 :size-28)}]]
 
-      :network
-      [tag-skeleton {:theme theme :size size :text network-name}
-       [rn/image {:style (style/circle-logo size) :source network-logo}]]
+        :network
+        [tag-skeleton {:theme theme :size size :text network-name}
+         [rn/image {:style (style/circle-logo size) :source network-logo}]]
 
-      :collectible
-      [tag-skeleton
-       {:theme       theme
-        :size        size
-        :text        (str collectible-name " #" collectible-number)
-        :shrinkable? true}
-       [rn/image {:style (style/rounded-logo size) :source collectible}]]
+        :collectible
+        (let [gray-text?       (string/blank? collectible-name)
+              collectible-text (cond
+                                 gray-text?         "UNKNOWN"
+                                 collectible-number (str collectible-name
+                                                         " #"
+                                                         collectible-number)
+                                 :else              collectible-name)
+              nft-placeholder? (or image-error? (string/blank? collectible))]
+          [tag-skeleton
+           {:theme       theme
+            :size        size
+            :text        collectible-text
+            :shrinkable? true
+            :gray-text?  gray-text?}
+           (if nft-placeholder?
+             [icons/icon :i/nft {:size 20}]
+             [rn/image {:style    (style/rounded-logo size)
+                        :source   collectible
+                        :on-error #(set-image-error true)}])])
 
-      :account
-      [tag-skeleton {:theme theme :size size :text account-name}
-       [account-avatar/view
-        {:customization-color customization-color
-         :emoji               emoji
-         :size                (if (= size 24) 20 28)}]]
+        :account
+        [tag-skeleton {:theme theme :size size :text account-name}
+         [account-avatar/view
+          {:customization-color customization-color
+           :emoji               emoji
+           :size                (if (= size 24) 20 28)}]]
 
-      :address
-      [address-tag props]
+        :address
+        [address-tag props]
 
-      :icon
-      [icon-tag props]
+        :icon
+        [icon-tag props]
 
-      :wallet-user
-      [tag-skeleton {:theme theme :size size :text full-name}
-       [wallet-user-avatar/wallet-user-avatar
-        {:full-name           full-name
-         :size                (if (= size 24) :size-20 :size-24)
-         :customization-color customization-color}]]
+        :wallet-user
+        [tag-skeleton {:theme theme :size size :text full-name}
+         [wallet-user-avatar/wallet-user-avatar
+          {:full-name           full-name
+           :size                (if (= size 24) :size-20 :size-24)
+           :customization-color customization-color}]]
 
-      :dapp
-      [tag-skeleton {:theme theme :size size :text dapp-name}
-       [rn/image {:style (style/circle-logo size) :source dapp-logo}]]
+        :dapp
+        [tag-skeleton {:theme theme :size size :text dapp-name}
+         [rn/image {:style (style/circle-logo size) :source dapp-logo}]]
 
-      nil)]])
+        nil)]]))
 
 (def view (schema/instrument #'view-internal component-schema/?schema))

--- a/src/status_im/contexts/wallet/common/activity_tab/view.cljs
+++ b/src/status_im/contexts/wallet/common/activity_tab/view.cljs
@@ -26,7 +26,8 @@
                           :collectible-name   (if (> amount 1)
                                                 (str amount " " nft-name)
                                                 nft-name)
-                          :collectible-number token-id}
+                          :collectible-number (when (not= token-id "0")
+                                                token-id)}
       :second-tag-prefix :t/from
       :second-tag        {:type :address :address sender}
       :third-tag-prefix  :t/to

--- a/src/status_im/subs/wallet/activities.cljs
+++ b/src/status_im/subs/wallet/activities.cljs
@@ -1,6 +1,5 @@
 (ns status-im.subs.wallet.activities
   (:require
-    [clojure.string :as string]
     [native-module.core :as native-module]
     [quo.foundations.resources :as quo.resources]
     [quo.foundations.resources]

--- a/src/status_im/subs/wallet/activities.cljs
+++ b/src/status_im/subs/wallet/activities.cljs
@@ -33,12 +33,6 @@
       (native-module/hex-to-number)
       (str)))
 
-(defn- normalize-nft-name
-  [token-id nft-name]
-  (if (and (some? token-id) (string/blank? nft-name))
-    "Unknown"
-    nft-name))
-
 (defn- get-token-amount
   [token amount]
   (let [token-type (:token-type token)]
@@ -77,7 +71,7 @@
 (defn- process-activity-by-type
   [chain-id->network-name
    {:keys [activity-type activity-status timestamp sender recipient token-in token-out
-           chain-id-in chain-id-out nft-name]
+           chain-id-in chain-id-out]
     :as   data}]
   (let [network-name (chain-id->network-name (or chain-id-in chain-id-out))
         token-id     (some-> (or token-in token-out)
@@ -91,8 +85,7 @@
                             :network-name  network-name
                             :token-id      token-id
                             :status        (constants/wallet-activity-status->name activity-status)
-                            :network-logo  (quo.resources/get-network network-name)
-                            :nft-name      (normalize-nft-name token-id nft-name))]
+                            :network-logo  (quo.resources/get-network network-name))]
     (condp = activity-type
       constants/wallet-activity-type-send
       (process-send-activity activity)

